### ISSUE TITLE
auto-detect raw format

### ISSUE THREAD

**Issue by lambdafu** [das-labor/panopticon#99]
Blue sky issue, but: there are going to be a lot of formats supported.  They need a better organization (hierarchical), but also, panopticon could do a better job by statistical analysis.  Surely every target architecture has common bit patterns. Just a touch of machine learning and we could do a nice thing here.

**Comment by flanfly** [comment]
I have plans to integrate a patter matching engine like YARA, PEiD, FLIRT, libmagic or binwalk that identifies file formats, packer and crypto constants.

ML for identifying the ISA of a dump you be cool but I don't have expertise in this field and getting the training data could be difficult.

**Comment by lambdafu** [comment]
Right, existing solutions based on pattern matching would be quite sufficient.
